src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -93,10 +93,12 @@ def __init__(
             engine_args=engine_args,
             context=engine_context,
         )
+
         self.sequence_length = sequence_length
         self.sampling_temperature = sampling_temperature
         self.deterministic = deterministic
         self.input_ids_length = input_ids_length
+        self.cache_length = sequence_length - input_ids_length
```

Comment:
Why do we subtract the input_ids_length here? Aren't the inputs stored in the cache?

Reply:
discussed in DMs